src/header.rs

```diff
@@ -357,8 +358,7 @@ impl TbfHeader {
             header_length += mem::size_of::<TbfHeaderKernelVersion>();
         }
 
-        // Flags default to app is enabled.
-        let flags = 0x0000_0001;
+        let flags = if disabled { 0x0000_0000 } else { 0x0000_0001 };
```

Comment:
Nit (and not really your error):  The enable/disable bit should be defined as a `const` or `bitflags!` or similar.  A similar definition should be used in [tock](https://github.com/tock/tock/blob/master/libraries/tock-tbf/src/types.rs#L569).